statannotations/Annotation.py

```diff
@@ -43,3 +44,12 @@ def print_labels_and_content(self, sep=" vs. "):
                                  for struct in self.structs)
 
         print(f"{labels_string}: {self.formatted_output}")
+
+    def check_data_stat_result(self):
+        if not isinstance(self.data, StatResult):
+            warnings.warn("Annotation data has incorrect class." +
+                          "Should be StatResult. Cannot annotate current pair.")
+            return False
+        return True
```

Comment:
I'm not sure this is necessary. As explained in the other comment, it should be straightforward that if you don't want to show non significant results, but you pass 'static' annotations, it's your job to choose which to pass.
And if we don't need to warn, there is no need for an extra method for checking `isinstance`, right?

Reply:
I opted for this implementation as it is easier to unit test and include a warning compared to what I originally had in mind. Given the logic above this is no longer necessary.